os/board/common/partitions.c

```diff
@@ -121,17 +123,25 @@ void configure_partitions(void)
 		}
 
 #if defined(CONFIG_MTD_PARTITION_NAMES)
-		if (*names) {
-			mtd_setpartitionname(mtd_part, names);
-		}
 
-		while (*names) {
-			if (*names == ',') {
+		if (*names) {
+			while (*names) {
+				if (*names == ',') {
+					part_name[index] = '\0';
+					index = 0;
+					names++;
+					break;
+				}
+				if (index > MTD_PARTNAME_LEN) {
+					lldbg("ERROR: Partition name is so long. Please make it smaller than %d\n", MTD_PARTNAME_LEN);
+					return;
```

Comment:
```
if (index <= MTD_PARTNAME_LEN) {
  part_name[index++] = *(names++);
} else {
  lldbg("ERROR: xxxxx);
  break;
}
```
This looks better. 
Because
when `index` become MTD_PARTNAME_LEN, next index is not needed anymore. But your codes work all of operation again, finally it just returns. It is redundant.

Reply:
And I suggested `break;` instead of `return;`. Because even if partition name is longer than buffer, you can use it with maximum size.
For example, partition name is "abcde" and buffer size is 3. Then, you can use "abc" as a partition name. You don't need to return without setting partition name.